src/middlewared/middlewared/plugins/service_/services/glusterd.py

```diff
@@ -27,3 +31,14 @@ async def before_stop(self):
 
     async def after_stop(self):
         await self.middleware.call('service.stop', 'glustereventsd')
+
+        # systemd[1]: glusterd.service: Unit process 1376703 (glusterfsd) remains running after unit stopped.
+        # systemd[1]: glusterd.service: Unit process 1376720 (glusterfs) remains running after unit stopped.
+        # This prevents from tank/.system/ctdb_shared_vol from being unmounted
+        futures = []
+        for process in psutil.process_iter(attrs=['cmdline']):
```

Comment:
Running synchronous code within the asyncio loop. Isn't there a more direct/smart way to do this other than loop through all processes?

Reply:
Gluster has a PID directory but now you're having to do IO on the asyncio loop which isn't any better. At this point it might just be worth it to `pkill -9 -f gluster`...